Feat-Edit-Delete

vinutha-101 opened the pull request.

vinutha-101 committed:
Create edit.html

vinutha-101 committed:
Create edit.js

The JS file for the edit and delete feature

vinutha-101 committed:
Update edit.html

vinutha-101 committed:
Update edit.html

Completed version of the edit.html file

vinutha-101 committed:
Update edit.html

vinutha-101 committed:
Update edit.js

vinutha-101 committed:
Update index.html

The updated version of index.html

vinutha-101 committed:
Create form.css

The CSS file for the form.html page.

vinutha-101 committed:
Create form.html

The HTML file for the form page.

vinutha-101 committed:
Create form.js

The JS file for the form page.

vinutha-101 committed:
Create table.js

The JS file for the index page(the table page)

vinutha-101 committed:
Delete style.css

Deleting the previous CSS file which is currently not needed.

vinutha-101 committed:
Delete display.html

vinutha-101 committed:
Delete display.js

vinutha-101 committed:
Delete script.js

vinothkumarsip commented on table.js:
```diff
@@ -0,0 +1,48 @@
+
+function deleteEntry(index) {
+    if (confirm("Are you sure you want to delete this entry?")) {
+        let entries = JSON.parse(localStorage.getItem("entries")) || [];
```
localStorage? but the requirement is about saving into json

maheprasath-ip commented on edit.html:
```diff
@@ -0,0 +1,72 @@
+<body>
+<form class="row g-3" id="edit-form">
+    <div class="container">
+        <div class="title" style="font-size:x-large;">Edit Entry</div>
```
why inline styling?  
what's the use of class="title" ?

vinutha-101 replied:
The inline styling was used since it was only for the title and so that it could be checked easily and the class="title" was used for the purpose of identifying the start of the content easily.

maheprasath-ip commented on form.html:
```diff
@@ -0,0 +1,75 @@
+  <form class="row g-3" id="data-form" method="POST">
+      
+      <div class="container">
+        <div class="title" style="font-size:x-large;">Application Form</div>
```
why ln line styling ?

maheprasath-ip commented on form.js:
```diff
@@ -0,0 +1,48 @@
+
+        console.log("Save button clicked!");
+
+        let name = document.getElementById("name").value.trim();
```
why let? are you going to modify the value?

maheprasath-ip commented on index.html:
```diff
@@ -1,82 +1,168 @@
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
+    <style>
+        body {
+            background: linear-gradient(135deg, #71b7e6, #9b59b6);
```
move this to a separate CSS file. I noticed the container style is used in two places let's combine them.

maheprasath-ip commented on index.html:
```diff
@@ -1,82 +1,168 @@
+            width: 90%;
+            max-width: 80vw;
+            background: #fff;
+            padding: 3%;
```
We should not use percentages for padding

maheprasath-ip commented on index.html:
```diff
@@ -1,82 +1,168 @@
+            margin: 0;
+        }
+
+        .container {
```
You are using Bootstrap, try use width style from Bootstrap

maheprasath-ip commented on index.html:
```diff
@@ -1,82 +1,168 @@
+        }
+
+        .btn-delete, .btn-edit{
+            width: 50%; 
```
for button, we don't use the %,  trying to control using padding

maheprasath-ip commented on form.js:
```diff
@@ -0,0 +1,48 @@
+
+        localStorage.setItem("entries", JSON.stringify(entries));
+
+        console.log("Data saved:", entries); 
```
remove the console when you give the PR to review

maheprasath-ip commented on form.js:
```diff
@@ -0,0 +1,48 @@
+    const form = document.getElementById("data-form");
+
+    if (!form) {
+        console.error("Form not found! Check if #data-form exists in form.html.");
```
Remove it

vinutha-101 committed:
Create db.json

The json file for the form entries

vinutha-101 committed:
Added json file

-The json file stores the entries from the form.
-The html file for the edit page(edit.html) includes a new id for save button
-The js files for edit page, table page and form page have been edited to get the data from the json file

vinothkumarsip commented on db.json:
```diff
@@ -0,0 +1,14 @@
+        "id": 1,
+        "name": "John Doe",
+        "age": 25,
+        "select": "BBA",
```
why this field is called "select"?

vinutha-101 committed:
Update with BootStrap

-The HTML files, form.html, edit.html and index.html, have been updated to make use of BootStrap
-The unnecessary parts of CSS file have been removed after including BootStrap
-The form.js now uses crypto.getRandomValues to generate unique IDs for each entry.
